### PR TITLE
TBI - refactor locate-reuse-lib.js template to use modern API's and fix lint issues

### PR DESCRIPTION
**Pull request by devinea**
TBI - https://github.com/SAP/open-ux-tools/issues/969
refactor locate-reuse-lib.js template to : 
- use modern UI5 API's 
- fix lint issues
- fix async loading

### CODE REVIEW

**Review by korotkovao — DISMISSED** [review]

**Review by IainSAP — DISMISSED** [review]
Thx @devinea 
- Tests updated as expected
- Did not test `locate-reuse-libs.js` changes, but verified that it was tested manually

File: packages/ui5-application-writer/templates/optional/loadReuseLibs/webapp/test/locate-reuse-libs.js [hunk 1]
```diff
@@ -100,45 +102,63 @@
...
     .finally(function () {
-        // setting the app title with internationalization
-        sap.ui.getCore().attachInit(bundleResources);
+
+        // setting the app title with internationalization 
+        sap.ui.getCore().attachInit(function () {
+            var sLocale = sap.ui.getCore().getConfiguration().getLanguage();
+            sap.ui.require(["sap/base/i18n/ResourceBundle"], function (ResourceBundle) {
+                var oResourceBundle = ResourceBundle.create({
+                    url: "i18n/i18n.properties",
```
Comment (ullasholla) [inline packages/ui5-application-writer/templates/optional/loadReuseLibs/webapp/test/locate-reuse-libs.js:156]: Typically, the resource bundles will be named `i18n/i18n.*properties`, but my understanding is that they don't _have_ to follow the naming convention. The authoritative place for the name would be the manifest. Do we need to handle the case where the resource bundle can't be located?

File: packages/ui5-application-writer/templates/optional/loadReuseLibs/webapp/test/locate-reuse-libs.js [hunk 2]
```diff
@@ -154,15 +174,22 @@ sap.registerComponentDependencyPaths(manifestUri)
...
                 sap.ui.require(["sap/ui/core/ComponentSupport"]);
 
-                // setting the app title with the i18n text
-                sap.ui.getCore().attachInit(bundleResources);
+                // setting the app title with the i18n text 
+                sap.ui.getCore().attachInit(function () {
+                    var sLocale = sap.ui.getCore().getConfiguration().getLanguage();
+                    sap.ui.require(["sap/base/i18n/ResourceBundle"], function (ResourceBundle) {
+                        var oResourceBundle = ResourceBundle.create({
+                            url: "i18n/i18n.properties",
```
Comment (ullasholla) [inline packages/ui5-application-writer/templates/optional/loadReuseLibs/webapp/test/locate-reuse-libs.js:182]: Same comment as https://github.com/SAP/open-ux-tools/pull/966#discussion_r1158829801

**Comment by ullasholla** [comment]
Does the file `locate-reuse-libs.js` need to turn strict mode on inside the functions? https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Strict_mode

File: packages/ui5-application-writer/templates/optional/loadReuseLibs/webapp/test/locate-reuse-libs.js [hunk 3]
```diff
@@ -26,72 +27,73 @@
...
-                            jQuery.sap.registerModulePath(dependency.componentId, dependency.url);
-                        }
-                    });
-                }
-            });
-        }
-    };
-
+    function registerModules(dataFromAppIndex) {
+        Object.keys(dataFromAppIndex).forEach(function (moduleDefinitionKey) {
```
[the remark this answers is not in the document]
Comment (ullasholla) [reply packages/ui5-application-writer/templates/optional/loadReuseLibs/webapp/test/locate-reuse-libs.js:74]: The check is done in the calling code: https://github.com/SAP/open-ux-tools/blob/e8b8cef411b7fbac9ef97525d0de4e385227c010/packages/ui5-application-writer/templates/optional/loadReuseLibs/webapp/test/locate-reuse-libs.js#L122-L123

Marking as resolved

**Review by ullasholla — CHANGES_REQUESTED** [review]
@devinea, thanks for the PR. The changes to `locate-reuse-libs.js` looks good. The tests are all correctly updated.

I have _not_ run the code and tested if it works as expected.

Had two questions:
* about using strict mode, https://github.com/SAP/open-ux-tools/pull/966#issuecomment-1497891444
* Handling resource bundle name mismatch (2 locations)

File: packages/ui5-application-writer/templates/optional/loadReuseLibs/webapp/test/locate-reuse-libs.js (hunk 2 above)
Comment (devinea) [reply packages/ui5-application-writer/templates/optional/loadReuseLibs/webapp/test/locate-reuse-libs.js:182]: see https://github.com/SAP/open-ux-tools/pull/966#discussion_r1158838857

**Comment by devinea** [comment]
> Does the file `locate-reuse-libs.js` need to turn strict mode on inside the functions? https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Strict_mode

@ullasholla good point. As discussed, I'll create a follow on PR for that next sprint.

File: packages/ui5-application-writer/templates/optional/loadReuseLibs/webapp/test/locate-reuse-libs.js (hunk 1 above)
Comment (ullasholla) [reply packages/ui5-application-writer/templates/optional/loadReuseLibs/webapp/test/locate-reuse-libs.js:156]: If they change the location/name, they'll have to update this file as well. Fair enough

**Review by ullasholla — APPROVED** [review]
All the review comments have been addressed. Thank you @devinea

**Comment by devinea** [comment]
Thanks @korotkovao @IainSAP @ullasholla

**Pull request merged**